Gem/Code/Source/Components/UI/UiSettingsComponent.cpp

```diff
@@ -258,40 +258,83 @@ namespace MultiplayerSample
         // Get the current fullscreen state. Unlike the other settings, we'll get this from the current window state so that we
         // handle things like Alt-enter that can change our windowing state regardless of what our user settings thinks.
 
-        // Start by defaulting to the user setting.
+        // Start by getting the user setting as our default state.
         bool fullscreen = false;
         MultiplayerSampleUserSettingsRequestBus::BroadcastResult(
             fullscreen, &MultiplayerSampleUserSettingsRequestBus::Events::GetFullscreen);
 
-        // Next, try to get the current state from the window. If it fails to get the state, we'll auto-default to the
+        // Next, try to get the current state from the window. If it fails to get the state, we'll default to the
         // user setting value that we fetched above.
-        AzFramework::WindowRequestBus::EventResult(fullscreen,
+        bool currentFullscreenState = fullscreen;
+        AzFramework::WindowRequestBus::EventResult(currentFullscreenState,
             GetWindowHandle(), &AzFramework::WindowRequestBus::Events::GetFullScreenState);
 
         // Rotate the index based on toggle direction.
-        uint32_t fullscreenIndex = GetRotatedIndex(valuesToLabels, fullscreen, toggleDirection);
+        uint32_t fullscreenIndex = GetRotatedIndex(valuesToLabels, currentFullscreenState, toggleDirection);
 
         UiTextBus::Event(toggle.m_labelEntity, &UiTextInterface::SetText, valuesToLabels[fullscreenIndex].second);
 
-        MultiplayerSampleUserSettingsRequestBus::Broadcast(
-            &MultiplayerSampleUserSettingsRequestBus::Events::SetFullscreen, valuesToLabels[fullscreenIndex].first);
+        // Only set & save the state if the value has changed. We might get a lot of spurious notifications here, so
```

Comment:
nit: if we no why we get supurious notifications may be good to add ie on Windows the OS appears to send this multiple times..

Reply:
Actually this comment is probably no longer true, and I probably don't need the extra safeguards. This was from a previous iteration where I hadn't added the OnFullscreenModeChanged() notification yet, so I was having to listen to other notifications that fired more often. I'll see if I can just remove this bit.

Reply:
Yep, it was safe to remove. :)